### ISSUE TITLE
Castlevania : Lords of Shadow Ultimate Edition, crash at launch (234080)

### ISSUE THREAD

**Issue by GriefNorth** [ValveSoftware/Proton#443]
Works fine for me. Pls add to whitelist. My specs - https://gist.github.com/GriefNorth/7f4820f319fda7e45ff44c5b83c20cd0

**Comment by Zero86Sk** [comment]
Can conform it works full speed.

It froze for me once when i alt+tabed douring a cutscene but that can be on my end.
I just restared from the last checkpoint. Nothing bad.

my steam info:
https://gist.github.com/Zero86Sk/d629f657c1ae861c36657cdf238a7765

**Comment by ghost** [comment]
Runs fine here, you have to double alttab to go into desktop. DS4 controller is being properly mapped.
https://gist.github.com/daasuharo/d83c92ac04a9bd0c5edfc359c3c5a4de

**Comment by kisak-valve** [comment]
Castlevania : Lords of Shadow Ultimate Edition - crash at start up

Issue transferred from https://github.com/ValveSoftware/Proton/issues/1960.
@darkpumpkins posted on 2018-11-13T22:51:34:

# Compatibility Report
- Name of the game with compatibility issues: Castlevania : Lords of Shadow Ultimate Edition, crash at launch
- Steam AppID of the game: 234080

## System Information
- GPU: GTX 1070 ti
- Driver/LLVM version: nvidia 396.54.09
- Kernel version: 4.18.0-11-generic
- Link to full system information report as [Gist](https://gist.github.com/):
[config.txt](https://github.com/ValveSoftware/Proton/files/2578429/config.txt)
- Proton version: 3.16-4beta and olders

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.


[steam-234080_chunk.log](https://github.com/ValveSoftware/Proton/files/2578444/steam-234080_chunk.log)
I had to take a piece of the log cause it reached more than 70Mb .... 

## Symptoms 
The game doesn't launch at all. i try with all version of proton, with a GTX 970 and a GTX 1070 ti on both gnome-shell or openbox Window manager.

## Reproduction
Just run the game with any version of proton.

**Comment by kisak-valve** [comment]
Found in the 2018-12-11 white list update.

**Comment by zzhiyi** [comment]
Tested with proton 3.16-4 and 3.16-16. It is working fine.  One line from the user log is particular interesting. 

 50214.785:0008:0009:warn:module:load_dll Failed to load module L"gameoverlayrenderer.dll"; status=c0000135

It's either a problematic installation or the module indeed failed to load on specific setup.

**Comment by kirgahn** [comment]
got the same issue on proton 4.2.3, Ubuntu 18.04

System Informaton
    GPU: RX480 8G
    Driver/LLVM version: Mesa 18.2.2, LLVM 7.0.0
    Kernel version: 4.18.0-15-generic
    Link to full system information report as Gist: [steam-234080.log](https://gist.github.com/kirgahn/55e3760beaa7068252364d56c216cec4)
    Proton version: 4.2.3

The issue seems pretty straightforward:
ERROR: ld.so: object '/home/kirgahn/.steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (**wrong ELF class: ELFCLASS64**): ignored.

Don't know why it complains about the library being compiled for amd_64, perhaps related to the prefixes wine_arch being x86_32?

Edit:
bit more in detail:

wine: Unhandled page fault on read access to 0x00000000 at address 0x432d65 (thread 0009), starting debugger...
346445.343:0008:0009:trace:seh:start_debugger Starting debugger "winedbg --auto 8 140"
ERROR: ld.so: object '/home/kirgahn/.steam/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
346446.139:0008:0009:trace:seh:call_stack_handlers handler at 0x7b4853f0 returned 1
**Unhandled exception: page fault on read access to 0x00000000 in 32-bit code (0x00432d65).**

This seems to crash proton

**Comment by zzhiyi** [comment]
Right. The dll ignored error is due to LD_PRELOAD with different architecture. This shouldn't cause a error besides a warning.

**Comment by kirgahn** [comment]
Is there a clean way to override the steam runtime LD_PRELOAD path?

I've tried to pass the 32bit gameoverlayrenderer.so using the steam launch properties with both LD_PRELOAD and LD_LIBRARY_PATH to no avail. 

The 64bit binary is fed to the game anyway, probably due to the fact that the amd_64 distro of the steam runtime is included *before* the game launch options.

**Comment by zzhiyi** [comment]
> Is there a clean way to override the steam runtime LD_PRELOAD path?
> 
There is no need. As I said, that will cause a warning and that's all it does. It's not the problem why this game crashed.

**Comment by darkpumpkins** [comment]
The game no longer crash at startup ( and seems to work pretty good ). 
=> Proton version 3.16-4 ( selected by valve )

**Comment by shoober420** [comment]
# Compatibility Report
- Name of the game with compatibility issues: Castlevania: Lords of Shadow - Ultimate Edition
- Steam AppID of the game: https://steamdb.info/app/234080/

## System Information
- GPU: GTX 760
- Driver/LLVM version: NVIDIA 440.64
- Kernel version: 5.5.9
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 5.0-4

LOG:
[steam-234080.log](https://github.com/ValveSoftware/Proton/files/4346634/steam-234080.log)


## Symptoms
Crash on startup, no window

**Comment by kisak-valve** [comment]
@shoober420, same assessment as https://github.com/ValveSoftware/Proton/issues/764#issuecomment-600590478.

**Comment by shoober420** [comment]
# Whitelist Request
- Name of the game to be whitelisted: Castlevania: Lords of Shadow - Ultimate Edition
- Steam AppID of the game: https://steamdb.info/app/234080/

## System Information
- GPU: GTX 760
- Driver/LLVM version: NVIDIA 440.64
- Distro version: Arch
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 5.0-4

LOG:
[steam-234080.log](https://github.com/ValveSoftware/Proton/files/4357883/steam-234080.log)

Works great out of the box.

**Comment by Mr-nUUb** [comment]
The game works like a charm with Proton 7.0-6 up to 10.0-4. I did not test older versions. Using Proton 11.0 and experimental (bleeding edge) the game emits an extremely loud buzzing noise that won't stop and increases in volume until the game is closed.

## System Information
GPU: AMD Radeon RX 7900XTX
Driver/LLVM version: Mesa 26.1.4
Distro version: CachyOS
Proton version: 10.0-4
Steam System Information: https://gist.github.com/Mr-nUUb/4aa8587c32a6a5dbdf8dc70e876eb7f2 (german)

**Comment by codeweaverwill** [comment]
@Mr-nUUb I haven't been able to reproduce the issue. Could you add `PROTON_LOG=+tid,+seh,+mmdevapi,+winmm,+driver,+msacm,+midi,+dsound,+dsound3d,+xaudio2,+xapofx,+dmusic,+mci,+pulse,+oss,+alsa,+coreaudio,+timestamp %command%` to the launch options, reproduce the issue, then upload the log from your home directory?

**Comment by Mr-nUUb** [comment]
@codeweaverwill Here's the log of Proton Experimental Bleeding Edge. I use a virtual 7.1 surround setup (pipewire filter chain) with an equalizer preset for my headset. Don't know if that matters.

[steam-234080.log](https://github.com/user-attachments/files/29767445/steam-234080.log)
